magento/magento2
# Initialize authentication popup modal only when needed

### Description (*)
This PR is somehow an evolution of the https://github.com/magento/magento2/pull/27686 in which I have found that it is not possible to completely skip initialization of authentication popup. That said, by adding this additional check, we are able to save a lot of CPU time for all of the shops that have guest checkout enabled.

### Related Pull Requests
https://github.com/magento/inventory/pull/3261
https://github.com/magento/partners-magento2ee/pull/428


Here are the profiles of before:
![auth-popup-before](https://user-images.githubusercontent.com/829189/97103431-c29a2500-16ac-11eb-8b4a-2508b5eef31d.png)
and after:
![auth-popup-after](https://user-images.githubusercontent.com/829189/97103435-c8900600-16ac-11eb-8a75-bb105d228b85.png)

### Manual testing scenarios (*)

1. `setModalElement` function is no longer called when guest checkout is enabled.
2. Authentication popup still shows properly when guest user tries to go to the checkout but guest checkout is disabled.

### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#30672: Initialize authentication popup modal only when needed
